activity.py

```diff
@@ -556,18 +556,16 @@ def _make_entry_widgets(self):
         self._entry_grid.attach(self.send_button, 2, 0, 1, 1)
         self.send_button.show()
 
-        self.clear_button = EventIcon(icon_name='clear',
-                                pixel_size=self._entry_height)
-        self.clear_button.connect('button-press-event', self._clear_button_cb)
-        self._entry_grid.attach(self.clear_button, 3, 0, 1, 1)
-        self.clear_button.show()
+        self._entry.set_icon_from_icon_name(Gtk.EntryIconPosition.SECONDARY, "clear")
+        self._entry.set_icon_sensitive(Gtk.EntryIconPosition.SECONDARY, True)
+        self._entry.connect('icon-press', self._clear_icon_cb)
 
         if not self.get_shared():
             self._entry.set_sensitive(False)
             self.smiley_button.set_sensitive(False)
             self.send_button.set_sensitive(False)
 
-    def _clear_button_cb(self,widget, event):
+    def _clear_icon_cb(self, entry, icon_pos, event):
         self._entry.props.text = ''
```

Comment:
Haven't tested yet, but it's a good idea to use the entry widget that's passed to the function instead of directly calling the one created by the class.

Reply:
thanks. And sorry about the wrong commit message. In fact, I have cheked the commit link, but it seemed that I didn't  totally understand it before.After looking at other PR, I understand now. Next time, I will try the suggested format.

But  I don't clearly know how to test this activity. I have tried to follow the development-environment doc and downloaded sugar-live-build-20201217.hybrid.iso, but I can't open it. Could you please give me some guidance? And I'm sorry for any inconvenience caused to you.

Reply:
Oh it's not a wrong commit message, it can use some improvement that's all. Nothing to be sorry about.

For future reference, it'll be really great if you have a dev environment setup so you can test your changes and thankfully Quozl has suggested how can get one.

The iso image you downloaded was meant to be installed and you can do that in a virtual machione of your choice, once installed you can update it to the latest Debian release, but if you're looking at modifying just Sugar activities then packaged Sugar would be best.